test/spec/SearchResults/initialization.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+var SearchParameters = require('../../../src/SearchParameters');
+var SearchResults = require('../../../src/SearchResults');
+
+test('processingTime should be the sum of all individual times', function() {
+  var result = new SearchResults(new SearchParameters(), [
+    {
+      processingTimeMS: 1
+    },
+    {
+      processingTimeMS: 1
+    }
+  ]);
+
+  expect(result.processingTimeMS).toBe(2);
+});
+
+test('processingTime should be NaN for one value undefined', function() {
+  var result = new SearchResults(new SearchParameters(), [
+    {
+      processingTimeMS: undefined
+    },
+    {
+      processingTimeMS: 1
+    }
+  ]);
+
+  expect(result.processingTimeMS).toBe(NaN);
```

Comment:
this returned `1` in the previous implementation, but I don't think the engine ever returns this anyway. Added the test for this comment.

Reply:
I think `processingTimeMS` can be `undefined` because the API says it can be filtered out: https://www.algolia.com/doc/api-reference/api-parameters/responseFields/?language=js#usage-notes

To be confirmed.

If yes, than we should at least add a check similar to the one sumBy had
https://github.com/lodash/lodash/blob/master/.internal/baseSum.js#L14

And then maybe rollback this test in that case ?

Reply:
the previous implementation would have returned 1 there, I can also go with `=== undefined`